test2/test2/Controllers/CountryModelsController.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.EntityFrameworkCore;
+using test2.Data;
+using test2.Models;
+
+namespace test2.Controllers
+{
+    public class CountryModelsController : Controller
+    {
+        private readonly CountryContext _context;
+
+        public CountryModelsController(CountryContext context)
+        {
+            _context = context;
+        }
+
+        // GET: CountryModels
+        public async Task<IActionResult> Index()
+        {
+            return View(await _context.CountryModel.ToListAsync());
+        }
+
+        // GET: CountryModels/Details/5
+        public async Task<IActionResult> Details(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var countryModel = await _context.CountryModel
+                .FirstOrDefaultAsync(m => m.Id == id);
+            if (countryModel == null)
+            {
+                return NotFound();
+            }
+
+            return View(countryModel);
+        }
+
+        // GET: CountryModels/Create
+        public IActionResult Create()
+        {
+            return View();
+        }
+
+        // POST: CountryModels/Create
+        // To protect from overposting attacks, enable the specific properties you want to bind to.
+        // For more details, see http://go.microsoft.com/fwlink/?LinkId=317598.
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Create([Bind("Id,Name,Salary,Area,People,Sea")] CountryModel countryModel)
+        {
+            if (ModelState.IsValid)
+            {
+                _context.Add(countryModel);
+                await _context.SaveChangesAsync();
+                return RedirectToAction(nameof(Index));
+            }
+            return View(countryModel);
+        }
+
+        // GET: CountryModels/Edit/5
+        public async Task<IActionResult> Edit(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var countryModel = await _context.CountryModel.FindAsync(id);
+            if (countryModel == null)
+            {
+                return NotFound();
+            }
+            return View(countryModel);
+        }
+
+        // POST: CountryModels/Edit/5
+        // To protect from overposting attacks, enable the specific properties you want to bind to.
+        // For more details, see http://go.microsoft.com/fwlink/?LinkId=317598.
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Edit(int id, [Bind("Id,Name,Salary,Area,People,Sea")] CountryModel countryModel)
+        {
+            if (id != countryModel.Id)
+            {
+                return NotFound();
+            }
+
+            if (ModelState.IsValid)
+            {

```

Comment:
Желательно избегать излишней вложенности, такой код сложнее читать
Общая практика:
```suggestion
            if (!ModelState.IsValid)
            {
                 return BadRequest();
            }
            // дальше идет код обновления модели
```
То есть мы проверили это условие с отрицанием и сразу отреагировали. Если модель валидна - пропускаем условие и идем дальше. Поведение тоже самое, но код имеет на 1 вложенность меньше.